crates/hyperqueue/src/dashboard/ui/screens/home/worker_info_table.rs

```diff
@@ -0,0 +1,102 @@
+use crate::dashboard::data::DashboardData;
+use crate::dashboard::ui::terminal::DashboardFrame;
+use crate::dashboard::ui::widgets::table::{StatefulTable, TableColumnHeaders};
+use crate::WorkerId;
+use tako::messages::common::WorkerConfiguration;
+use tui::layout::{Constraint, Rect};
+use tui::widgets::{Cell, Row};
+
+#[derive(Default)]
+pub struct WorkerInfoTable {
+    table: StatefulTable<WorkerInfoDataRow>,
+    current_worker_id: Option<WorkerId>,
+}
+
+#[derive(Default, Debug)]
+struct WorkerInfoDataRow {
+    pub label: &'static str,
+    pub data: String,
+}
+
+impl WorkerInfoTable {
+    pub fn update(&mut self, data: &DashboardData, worker_id: Option<WorkerId>) {
+        self.current_worker_id = worker_id;
+        self.table.set_items(vec![]);
+        if let Some(worker_id) = self.current_worker_id {
+            let info = data.query_worker_info_for(&worker_id);
+            if let Some(info) = info {
+                let rows = create_rows(info);
+                self.table.set_items(rows);
+            }
+        }
+    }
+
+    pub fn draw(&mut self, rect: Rect, frame: &mut DashboardFrame) {
+        self.table.draw(
+            rect,
+            frame,
+            TableColumnHeaders {
+                title: "Worker Info".to_string(),
+                inline_help: "".to_string(),
+                table_headers: None,
+                column_widths: vec![Constraint::Percentage(30), Constraint::Percentage(70)],
+            },
+            |data| Row::new(vec![Cell::from(data.label), Cell::from(data.data.clone())]),
```

Comment:
```suggestion
            |data| Row::new(vec![Cell::from(data.label), Cell::from(data.data.as_str())]),
```
Now we can avoid the clone :)